src/js/components/shepherd-element/index.jsx

```diff
@@ -65,8 +65,13 @@ export default class ShepherdElement extends Component {
     const { tour } = this.step;
     switch (e.keyCode) {
       case KEY_TAB:
+        if (this.focusableElements.length === 0) {
+          e.preventDefault();
+          break;
+        }
         if (this.focusableElements.length === 1) {
```

Comment:
Could we remove this case? Essentially just have the base case of no focusable elements, and otherwise do the normal tabbing. I believe it should work without having a case for a single element. Do you mind trying it?

Also, I am curious how would you end up with a tour element with no focusable elements? Are the controls for advancing the tour outside of the tooltip?

Reply:
> Could we remove this case? Essentially just have the base case of no focusable elements, and otherwise do the normal tabbing. I believe it should work without having a case for a single element. Do you mind trying it?

This is for the case there are no focusable elements (tried it before the commit). what happens is when user clicks TAB, the focus leaves the tour and moves outside the step.

> Also, I am curious how would you end up with a tour element with no focusable elements? Are the controls for advancing the tour outside of the tooltip?

We have a step that we ask user to perform an action ("Click here to ...."). once he clicks, we advance to next step programatically. The step is minimal without option to exit the tour and with no buttons.
(we rarely do it but there are a few cases that are critical so we require users to take action)

Reply:
Regarding accessibility we are now in brainstorming of how to solve it actually. Since the step requires user to perform action, to perform it user needs to have focus on the clickable element, and there is no such support it shepherd for that.

Reply:
> This is for the case there are no focusable elements (tried it before the commit). what happens is when user clicks TAB, the focus leaves the tour and moves outside the step.

I'm not sure we are on the same page here. I am saying to keep the no focusable elements case, which is the `0` case, and remove the `1` case in favor of the default functionality. It should work.

> Regarding accessibility we are now in brainstorming of how to solve it actually. Since the step requires user to perform action, to perform it user needs to have focus on the clickable element, and there is no such support it shepherd for that.

This is a separate issue. We should probably allow tabbing out of the tour, if `useModalOverlay` is false, but that would be a separate thing from fixing this.

Reply:
@rwwagner90 agree, it's a separate issue.

as a side note: since the step is a dialog if I'm not mistaken, according to accessibility standards it should not allow tabbing out, regardless if there is an overlay.

Reply:
@genadis yes, that is why it was originally set to not allow tabbing out at all. I'm open to suggestions on how to make things work for both of us, while still adhering to a11y standards.

Reply:
@rwwagner90 how do you feel about adding an event that fires if user activates a step element - `space` or `enter` when the focus is on the step element.

We can then leverage this event in our app:
1. in our step that requires the user to click without the next button, we will add hidden accessibility message asking user to use the space or enter to move to next step.
2. catch the event, do the action programatically and proceed to next step.

what do you think?

Reply:
actually we can catch the keboard event in our app directly as well, so no need.

thanks

Reply:
I'm not sure I quite follow, but if you can handle it in your app, that is great! Please let me know if you end up needing anything on the Shepherd end.